app/vmctl/remoteread/remoteread.go

```diff
@@ -0,0 +1,263 @@
+package remoteread
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/VictoriaMetrics/VictoriaMetrics/lib/bytesutil"
+	"github.com/VictoriaMetrics/VictoriaMetrics/lib/logger"
+	"github.com/gogo/protobuf/proto"
+	"github.com/golang/snappy"
+	"github.com/prometheus/prometheus/model/labels"
+	"github.com/prometheus/prometheus/prompb"
+	"github.com/prometheus/prometheus/storage/remote"
+	"github.com/prometheus/prometheus/tsdb/chunkenc"
+)
+
+var bodyBufferPool bytesutil.ByteBufferPool
+
+const (
+	defaultReadTimeout = 30 * time.Second
+	remoteReadPath     = "/api/v1/read"
+	healthPath         = "/health"
+)
+
+// StreamCallback is a callback function for processing time series
+type StreamCallback func(series prompb.TimeSeries) error
+
+// Client is an HTTP client for reading
+// time series via remote read protocol.
+type Client struct {
+	addr     string
+	c        *http.Client
+	user     string
+	password string
+}
+
+// Config is config for remote read.
+type Config struct {
+	// Addr of remote storage
+	Addr string
+	// ReadTimeout defines timeout for HTTP write request
+	// to remote storage
+	ReadTimeout time.Duration
+	// Username is the remote read username, optional.
+	Username string
+	// Password is the remote read password, optional.
+	Password string
+}
+
+// Filter is used for request remote read data by filter
+type Filter struct {
+	Min, Max   int64
```

Comment:
Min, Max are bad names, I cannot understand what does it mean without reading all code.

Reply:
renamed